### ISSUE TITLE
Spin Configuration of Individual roots in CASSCF calculations

### ISSUE THREAD

**Issue by BallisticBill** [pyscf/pyscf#1946]
I was trying out a CASSCF(4,4) calculation on the first 6 roots of the trimethylenemethane biradical. I wanted to find the spin configuration of each individual root, but the simulation output did not contain such information. For example running the same calculation in ORCA gave the following output.

```
ROOT   0:  E=    -154.8651120698 Eh
      0.92750 [     1]: 2110
      0.07249 [     8]: 1120
ROOT   1:  E=    -154.8476994989 Eh  0.474 eV   3821.6 cm**-1
      0.46838 [     0]: 2200
      0.33407 [     3]: 2020
      0.19723 [     6]: 1210
ROOT   2:  E=    -154.7383675967 Eh  3.449 eV  27817.2 cm**-1
      0.56321 [     3]: 2020
      0.37203 [     0]: 2200
      0.06244 [    13]: 0220
ROOT   3:  E=    -154.6774350968 Eh  5.107 eV  41190.3 cm**-1
      0.90286 [     4]: 2011
      0.04178 [     7]: 1201
      0.03460 [    14]: 0211
      0.02076 [    11]: 1021
ROOT   4:  E=    -154.6716383953 Eh  5.265 eV  42462.6 cm**-1
      0.79779 [     2]: 2101
      0.18405 [     9]: 1111
      0.01816 [    16]: 0121
ROOT   5:  E=    -154.5426490243 Eh  8.775 eV  70772.5 cm**-1
      0.80046 [     6]: 1210
      0.13694 [     0]: 2200
      0.06235 [     3]: 2020
```
where adjacent to the weight of each root, the information on the occupancy is also given. Is there a method to obtain this in PySCF? I wanted to use this information to find out the coupling of the spins.

**Comment by MatthewRHermes** [comment]
I'm a bit confused, you say "spin configuration" and yet I don't see that information in your ORCA output. The S^2 expectation value for each root should be printed next to its energy; for example,
```
CASCI energy for each state
  State 0 weight 0.5  E = -113.910222700617 S^2 = 0.0000000
  State 1 weight 0.5  E = -113.737744347562 S^2 = 0.0000000
```
As for CI vector coefficients/weights from your ORCA printout, if you run `mc.analyze ()` after `mc.kernel ()` (where `mc` is your CASSCF method object), then beginning with the line `** Largest CI components **` in your output file there should be a printout of large CI vector coefficients.

**Comment by BallisticBill** [comment]
By the spin configuration I mean that I can see how the active space is populated in the ORCA output. For example in the case of Root 0, following the weight of CI vector, "2110" indicates that the lowest energy orbital in doubly occupied while the other two are singly occupied leaving the highest orbital empty. This is how I interpreted the data. Please correct me if I am wrong.

**Comment by MatthewRHermes** [comment]
Yeah that information is contained in the `** Largest CI components **` block I pasted above, it's just in a different format. The three rows above correspond to "2200", "0220", and "2002".

**Comment by hebrewsnabla** [comment]
I have a small script to translate PySCF's configuration to ORCA style which may help you

https://github.com/hebrewsnabla/pyAutoMR/blob/27cfefeab881f766388203943f5110a60000925c/automr/cidump.py#L25

**Comment by BallisticBill** [comment]
@hebrewsnabla Thank you!. This was very helpful.

**Comment by verena-neufeld** [comment]
Hi @BallisticBill , thanks for your question. Has it been answered?
